src/main/java/net/neoforged/neoforge/event/EventHooks.java

```diff
@@ -797,8 +797,8 @@ public static long onSleepFinished(ServerLevel level, long newTime, long minTime
         return event.getNewTime();
     }
 
-    public static List<PreparableReloadListener> onResourceReload(ReloadableServerResources serverResources, RegistryAccess registryAccess) {
-        AddReloadListenerEvent event = new AddReloadListenerEvent(serverResources, registryAccess);
+    public static List<PreparableReloadListener> onResourceReload(ReloadableServerResources serverResources, RegistryAccess registryAccess, List<PreparableReloadListener> listeners) {
+        AddReloadListenerEvent event = new AddReloadListenerEvent(serverResources, registryAccess, listeners);
```

Comment:
Per the compatibility check, you'll need to reinstate the old two-args method for binary compatibility. However, since I think no mod will need to call this method specifically, the old method should just call this new method with an empty list, and this new method should be annotated with `@ApiStatus.Internal`.